video_worker/tests/test_video_worker.py

```diff
@@ -0,0 +1,535 @@
+            'jobid': 'dummy-job-id',
+            'workdir': '',
+            'instance_yaml': 'dummy-instance_yaml.yaml',
+        },
+        {
+            'jobid': 'dummy-job-id',
+            'workdir': '/dummy-work-dir',
+            'instance_yaml': 'dummy-instance_yaml.yaml',
+        }
+    )
+    def test_video_worker(self, data):
+        """
+        Tests video worker object is created with correct values.
+        """
+        VW = VideoWorker(**data)
+
+        # Verify instance_yaml has correct value
+        expected_instance_yaml = data.get(
+            'instance_yaml',
+            os.path.join(
+                os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                'instance_config.yaml'
+            )
+        )
+        self.assertEqual(VW.instance_yaml, expected_instance_yaml)
+
+        # Verify that workdir has correct value
+        job_id = data.get('jobid', None)
+        expected_workdir = data.get(
+            'workdir',
+            os.path.join(ENCODE_WORK_DIR, job_id) if job_id else ENCODE_WORK_DIR
+        )
+        self.assertEqual(VW.workdir, expected_workdir)
+
+    @data(True, False)
+    @patch('nose.run')
+    def test_video_worker_test(self, nose_result, mock_nose):
+        """
+        Test `test` method works correctly.
+        """
+        # Mock nose.run
+        mock_nose.return_value = nose_result
+        result = self.VW.test()
+        self.assertEqual(result, nose_result)
+
+    @data(
+        (
+            {
+                'error_message': 'No Encode Profile Specified'
+            }
+        ),
+        (
+            {
+                'is_valid': False,
+                'encode_profile': 'static-pipeline',
+                'error_message': 'Invalid Video / VEDA Data'
+            }
+        ),
+        (
+            {
+                'is_valid_engine_intake': False,
+                'encode_profile': 'static-pipeline',
+                'error_message': 'Invalid Video / Local'
+            }
+        ),
+        (
+            {
+                'encode_profile': 'static-pipeline',
+                'path_exists': False
+            }
+        ),
+        # Success
+        (
+            {
+                'encode_profile': 'hls'
+            }
+        ),
+        (
+            {
+                'encode_profile': 'static-pipeline'
+            }
+        )
+    )
+    @patch.object(VideoWorker, '_static_pipeline')
+    @patch.object(VideoWorker, '_hls_pipeline')
+    @patch.object(VideoWorker, '_update_api')
+    @patch('video_worker.video_images.VideoImages.settings_setup')
+    @patch('video_worker.video_images.VideoImages.create_and_update')
+    @patch.object(deliverable_route, 'apply_async')
+    @patch.object(video_worker_logger, 'error')
+    @patch('shutil.rmtree')
+    @patch('os.path.exists')
+    @patch('os.mkdir')
+    def test_run(self, mock_data, mock_mkdir, mock_exists, mock_rmtree, mock_logger, celeryapp_sync_mock,
```

Comment:
It is very hard to understand these big tests with so many patches. Can we make them simple?